app/src/main/java/org/sopt/and/presentation/authentication/SignInScreen.kt

```diff
@@ -8,44 +8,117 @@ import androidx.compose.foundation.layout.height
+import androidx.lifecycle.repeatOnLifecycle
 import org.sopt.and.R
-import org.sopt.and.presentation.authentication.components.CustomTextField
+import org.sopt.and.presentation.authentication.components.AlertText
+import org.sopt.and.presentation.authentication.components.AuthTextField
 import org.sopt.and.presentation.authentication.components.SignInTopBar
 import org.sopt.and.presentation.authentication.components.SnsAccountTab
+import org.sopt.and.presentation.common.CustomConfirmDialog
 import org.sopt.and.presentation.extension.noRippleClickable
 import org.sopt.and.presentation.theme.ANDANDROIDTheme
+import org.sopt.and.presentation.theme.Error
 import org.sopt.and.presentation.theme.LightGray
 import org.sopt.and.presentation.theme.WavveMain
-import org.sopt.and.presentation.util.Constants
 
 @Composable
 fun SignInScreen(
-    modifier: Modifier = Modifier,
-    onSignInClick: (String, String) -> Unit = { _, _ -> },
-    onSignUpClick: () -> Unit = {},
+    onNavigateToHome: (String?) -> Unit,
+    onNavigateToSignUp: () -> Unit,
+    viewModel: SignInViewModel = hiltViewModel()
+) {
+    val uiState by viewModel.uiState.collectAsState()
+    val lifecycleOwner = LocalLifecycleOwner.current
+    val signInMessage = stringResource(R.string.sign_in_success)
 
-    ) {
-    var emailInput by remember { mutableStateOf("") }
-    var passwordInput by remember { mutableStateOf("") }
+    var showEmailError by remember { mutableStateOf(false) }
+    var showPasswordError by remember { mutableStateOf(false) }
+
+    LaunchedEffect(viewModel.sideEffect, lifecycleOwner) {
+        lifecycleOwner.lifecycle.repeatOnLifecycle(Lifecycle.State.STARTED) {
+            viewModel.sideEffect.collect { sideEffect ->
+                when (sideEffect) {
+                    is SignInSideEffect.NavigateToHome -> onNavigateToHome(signInMessage)
+                    is SignInSideEffect.NavigateToSignUp -> onNavigateToSignUp()
+                    is SignInSideEffect.InvalidEmail -> {
+                        showEmailError = true
+                        showPasswordError = false
+                    }
+                    is SignInSideEffect.InvalidPassword -> {
+                        showPasswordError = true
+                        showEmailError = false
+                    }
+                    is SignInSideEffect.SignInFailed -> {
+                        showEmailError = false
+                        showPasswordError = false
+                    }
+                }
+            }
+        }
+    }
+
+    if (uiState.isDialogShown) {
+        CustomConfirmDialog(
+            title = R.string.wavve,
+            description = R.string.sign_in_failed,
+            onDismissRequest = {
+                viewModel.updateDialogVisibility(false)
+            },
+            dismissText = R.string.confirm,
+        )
+    }
+
+    SignInScreenContent(
+        emailInput = uiState.emailInput,
+        onEmailChange = { viewModel.updateEmailInput(it) },
```

Comment:
저는 람다가 있으면 괜히 거슬려서 필요가 없다면 쓰지 않는 편이에요!
람다 파라미터가 없거나 람다 파라미터와 함수의 매개변수의 타입과 순서가 일치한다면 람다를 제거하고 참조연산자 `::`를 사용하는 방법이 있답니다!
```suggestion
        onEmailChange = viewModel::updateEmailInput,
```

Reply:
반영해두겠습니다!!!